### PR TITLE
some (mostly) client/gui mappings

### PR DESCRIPTION
**Pull request by moehreag**

### CODE REVIEW

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/client/gui/Element.mapping [hunk 1]
```diff
@@ -137,5 +137,5 @@ CLASS net/minecraft/unmapped/C_fxiqpxaf net/minecraft/client/gui/Element
 			COMMENT the X coordinate of the mouse
 		ARG 3 mouseY
 			COMMENT the Y coordinate of the mouse
-		ARG 5 amount
-			COMMENT value is {@code < 0} if scrolled down, {@code > 0} if scrolled up
+		ARG 5 amountX
+		ARG 7 amountY
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/Element.mapping]: I'd add the comment back, and include it for the y amount

File: mappings/net/minecraft/client/gui/RenderList.mapping [hunk 2]
```diff
@@ -0,0 +1,17 @@
+CLASS net/minecraft/unmapped/C_nihbhmxv net/minecraft/client/gui/RenderList
+	FIELD f_afqpfbiz layers Ljava/util/List;
+	FIELD f_lrobklrs LAYER_OFFSET F
+	METHOD m_aolowtyu (Ljava/util/function/BooleanSupplier;Lnet/minecraft/unmapped/C_nihbhmxv;Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 3 delta
+	METHOD m_fyitalyz renderLayers (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/RenderList.mapping]:
```suggestion
		ARG 3 tickDelta
	METHOD m_fyitalyz renderLayers (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/RenderList.mapping [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	METHOD m_moqgrgyg add (Lnet/minecraft/unmapped/C_nihbhmxv$C_lkrghfli;)Lnet/minecraft/unmapped/C_nihbhmxv;
+		ARG 1 layer
+	METHOD m_odhxuxlp render (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/RenderList.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/RenderList.mapping [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	METHOD m_tqkbsprw add (Lnet/minecraft/unmapped/C_nihbhmxv;Ljava/util/function/BooleanSupplier;)Lnet/minecraft/unmapped/C_nihbhmxv;
+		ARG 1 subList
+		ARG 2 renderCondition
+	CLASS C_lkrghfli Layer
+		METHOD render render (Lnet/minecraft/unmapped/C_sedilmty;F)V
+			ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/RenderList.mapping]:
```suggestion
			ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	FIELD f_wnubqcju storage [Ljava/lang/Object;
+	METHOD <init> (I)V
+		ARG 1 size
+	METHOD add add (ILjava/lang/Object;)V
+		ARG 1 index
+		ARG 2 object
+	METHOD addFirst addFirst (Ljava/lang/Object;)V
+		ARG 1 object
+	METHOD addLast addLast (Ljava/lang/Object;)V
+		ARG 1 object
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping]: I feel like these `object`s could be a bit more specific

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 6]
```diff
@@ -62,7 +96,12 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_bkadrkcb shouldShowChatDisabledScreen ()Z
 	METHOD m_btqrbwce renderCrosshair (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
 	METHOD m_ciysbohv getSpectatorHud ()Lnet/minecraft/unmapped/C_ybztcyiy;
+	METHOD m_ckjzczem (Lnet/minecraft/unmapped/C_lfemghur;)Z
+		ARG 0 entry
+	METHOD m_cwobkefs renderOverlayMessage (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
	METHOD m_ciysbohv getSpectatorHud ()Lnet/minecraft/unmapped/C_ybztcyiy;
	METHOD m_ckjzczem (Lnet/minecraft/unmapped/C_lfemghur;)Z
		ARG 0 entry
	METHOD m_cwobkefs renderOverlayMessage (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 7]
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 2 tickDelta
 	METHOD m_fmuufkwf renderDemoTimer (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 8]
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
 	METHOD m_fvwzoqkl getHeartCount (Lnet/minecraft/unmapped/C_usxaxydn;)I
 		ARG 1 entity
 	METHOD m_fydpkivs setSubtitle (Lnet/minecraft/unmapped/C_rdaqiwdt;)V
 		ARG 1 subtitle
+	METHOD m_gbnvpsmk renderSleep (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
 	METHOD m_gyqjeoze getPlayerListHud ()Lnet/minecraft/unmapped/C_lybmfibg;
+	METHOD m_hmpeaweq renderChat (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
	METHOD m_gyqjeoze getPlayerListHud ()Lnet/minecraft/unmapped/C_lybmfibg;
	METHOD m_hmpeaweq renderChat (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 9]
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
 	METHOD m_hxolcuvy setOverlayMessage (Lnet/minecraft/unmapped/C_rdaqiwdt;Z)V
 		ARG 1 message
 		ARG 2 tinted
 	METHOD m_iimewzba getBossBarHud ()Lnet/minecraft/unmapped/C_vpkfuxkl;
+	METHOD m_irzrqyso getDebugHud ()Lnet/minecraft/unmapped/C_epxplegp;
+	METHOD m_iucytvnr renderScoreboard (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 10]
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 5 color
 	METHOD m_koznghsj renderAutosaveIndicator (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 11]
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
 	METHOD m_kukerlrn setTitleTicks (III)V
 		ARG 1 fadeInTicks
 		ARG 2 remainTicks
 		ARG 3 fadeOutTicks
+	METHOD m_lcsxraft renderPlayerList (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 12]
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
+	METHOD m_mocrhtmt (Lnet/minecraft/unmapped/C_ymwyplge;Lnet/minecraft/unmapped/C_zhwspcux;Lnet/minecraft/unmapped/C_lfemghur;)Lnet/minecraft/unmapped/C_nepwsmgw$C_utdjjnqv;
+		ARG 3 entry
 	METHOD m_nfjwrptj getTextRenderer ()Lnet/minecraft/unmapped/C_mavozmpp;
 	METHOD m_oafggdwi renderStatusEffectOverlay (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 13]
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_pwrfrppi getChatHud ()Lnet/minecraft/unmapped/C_uedmlrsy;
 	METHOD m_qblpluah updateVignetteDarkness (Lnet/minecraft/unmapped/C_astfners;)V
 		ARG 1 entity
+	METHOD m_qgtzabmx renderTitles (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
+	METHOD m_ronzofwm renderOverlays (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
	METHOD m_ronzofwm renderOverlays (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 14]
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
+		ARG 6 blinking
+		ARG 7 half
 	METHOD m_ruxdwerw renderPortalOverlay (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
 		ARG 2 nauseaStrength
 	METHOD m_shmqmkfw renderStatusBars (Lnet/minecraft/unmapped/C_sedilmty;)V
 		COMMENT Renders the armor, health, air, and hunger bars.
 		ARG 1 graphics
+	METHOD m_srkhzgsd renderHotbar (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 15]
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
+	METHOD m_ssocipmh (I)[Lnet/minecraft/unmapped/C_nepwsmgw$C_utdjjnqv;
+		ARG 0 size
 	METHOD m_twidsnlh getRiddenEntity ()Lnet/minecraft/unmapped/C_usxaxydn;
+	METHOD m_unxvdhni (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 16]
```diff
@@ -144,11 +208,17 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 1 title
 	METHOD m_wjrmovwk tickPausable ()V
 	METHOD m_wkdzfhws setDefaultTitleFade ()V
+	METHOD m_wscbnfkt (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
+	METHOD m_wttpxmtb (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
	METHOD m_wttpxmtb (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 17]
```diff
@@ -144,11 +208,17 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
 	METHOD m_wwnerxsu renderMountHealth (Lnet/minecraft/unmapped/C_sedilmty;)V
 		ARG 1 graphics
 	METHOD m_xgoqlosf tickAutosaveIndicator ()V
 	METHOD m_xqvrdrrt getHeartRows (I)I
 		ARG 1 heartCount
+	METHOD m_yahflzho renderSurvivalHotbar (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping]:
```suggestion
		ARG 2 tickDelta
```

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping [hunk 18]
```diff
@@ -161,5 +231,18 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
...
 	CLASS C_mdshohhj HeartType
+		FIELD f_chhxrkvp hardcoreHalfBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_ghyqxqvl hardcoreHalf Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_kedkcssv full Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_mapfxoof halfBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_mlgfossm half Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_rphpsjvq hardcoreFullBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_slrltipg hardcoreFull Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_wcxevweg fullBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		METHOD <init> (Ljava/lang/String;ILnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;)V
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:248]: Empty method

File: mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping [hunk 19]
```diff
@@ -84,6 +88,8 @@ CLASS net/minecraft/unmapped/C_nmdfcqgq net/minecraft/client/gui/widget/list/Ent
 		ARG 1 graphics
 		ARG 2 x
 		ARG 3 y
+	METHOD m_tpmgwusc isZero (I)Z
+		ARG 1 i
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping]: `index`?

File: mappings/net/minecraft/client/network/ServerInfo.mapping [hunk 20]
```diff
@@ -13,17 +14,22 @@ CLASS net/minecraft/unmapped/C_xotldzgg net/minecraft/client/network/ServerInfo
...
 	METHOD m_ijmgdohe setFavicon ([B)V
 		ARG 1 favicon
+	METHOD m_ilxbxxlb getPingResult ()Lnet/minecraft/unmapped/C_xotldzgg$C_odxprzma;
 	METHOD m_kiyulysu isRealm ()Z
 	METHOD m_ktfqzyqf getFavicon ()[B
 	METHOD m_mgdrmoly toNbt ()Lnet/minecraft/unmapped/C_hhlwcnih;
 	METHOD m_ntjhtcjm copyWithSettingsFrom (Lnet/minecraft/unmapped/C_xotldzgg;)V
 		ARG 1 serverInfo
+	METHOD m_rvsteyeu setResult (Lnet/minecraft/unmapped/C_xotldzgg$C_odxprzma;)V
+		ARG 1 result
```
Comment (IotaBread) [inline mappings/net/minecraft/client/network/ServerInfo.mapping]:
```suggestion
	METHOD m_rvsteyeu setPingResult (Lnet/minecraft/unmapped/C_xotldzgg$C_odxprzma;)V
		ARG 1 pingResult
```

**Review by IotaBread — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping (hunk 5 above)
Comment (moehreag) [reply mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping]: would you prefer `element`? That's what `java.util.AbstractList` uses.

Comment (IotaBread) [reply mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping]: That sounds good

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping (hunk 18 above)
Comment (moehreag) [reply mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:248]: I'm not entirely sure, do you mean the constructor? It's not empty, it just only assigns the fields to the parameters passed.

File: mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping (hunk 19 above)
Comment (moehreag) [reply mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping]: The method body is just `return i==0` so I'm not sure what to name the parameter really..

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping (hunk 18 above)
Comment (IotaBread) [reply mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:248]: I meant the mapping, it doesn't have a name or parameters

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping (hunk 18 above)
Comment (moehreag) [reply mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:248]: the parameter names were marked as suggestions (which I'm not entirely sure when they are put into the mappings file and when they aren't). I'm relatively certain that it's rather normal for a constructor to not have a mapped name.

**Pull request merged**